### PR TITLE
Convert project names to kebab case format

### PR DESCRIPTION
**Pull request by donat**
The "idiomatic" project naming in a Gradle build is to use kebab case formatting. The `gradle/gradle` build, however, converted the project names from the kebab case subproject directory names to camel case project names in order to use the abbreviated name matching feature. The latest Gradle snapshot now supports kebab case names in the name matching, so there's no need for the conversion.

### CODE REVIEW

**Comment by donat** [comment]
@blindpirate This project rename seems to have an effect on the CI build pipeline. Running [quick feedback](https://builds.gradle.org/viewLog.html?buildId=37225860&tab=buildResultsDiv&buildTypeId=Gradle_Check_Quick_1_bucket50) executes much more tests than usual. I'm assuming this is due to the new project names not being recognized properly by the CI bucket distribution algorithm. Can you please take a look?

File: buildSrc/subprojects/packaging/src/main/kotlin/gradlebuild/shaded-jar.gradle.kts [hunk 1]
```diff
@@ -95,13 +95,14 @@ fun createConfigurationToShade() = configurations.create("jarsToShade") {
...
             this.add(project.dependencies.create(project))
-            this.add(project.dependencies.create(project.dependencies.platform(project(":distributionsDependencies"))))
+            this.add(project.dependencies.create(project.dependencies.platform(project(":distributions-dependencies"))))
         }
     }
 
 fun addShadedJarTask(shadedJarExtension: ShadedJarExtension): TaskProvider<ShadedJar> {
     val configurationToShade = shadedJarExtension.shadedConfiguration
 
+    // TODO (donat) convert project name to kebab-case before prepending it to the task name
```
Comment (jjohannes) [inline buildSrc/subprojects/packaging/src/main/kotlin/gradlebuild/shaded-jar.gradle.kts:105]: Would be nice if these TODOs can be addressed directly (this PR, or a direct follow up). :)

File: buildSrc/subprojects/build-update-utils/src/main/kotlin/gradlebuild/buildutils/tasks/GenerateSubprojectsInfo.kt [hunk 2]
```diff
@@ -42,6 +42,7 @@ abstract class GenerateSubprojectsInfo : DefaultTask() {
 
     private
     fun generateSubproject(subprojectDir: File): GradleSubproject {
+        // TODO (donat) we should probably change this one as well
         return GradleSubproject(subprojectDir.name,
             subprojectDir.name.kebabToCamel(),
```
Comment (jjohannes) [inline buildSrc/subprojects/build-update-utils/src/main/kotlin/gradlebuild/buildutils/tasks/GenerateSubprojectsInfo.kt:47]: I think you need to remove the `kebabToCamel()` and regenerate the `subprojects.json` in order for CI to work correctly. You already adjusted the names in `lifecycle.gradle.kts`. The names are retrieved from the file that is generated by the code here I think.

**Comment by donat** [comment]
@jjohannes I've addressed all your review items.

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Pull request merged**